src/Raven.Studio/typescript/commands/database/documents/saveDocumentCommand.ts

```diff
@@ -5,26 +5,37 @@ import endpoints = require("endpoints");
 
 class saveDocumentCommand extends commandBase {
 
-    constructor(private id: string, private document: document, private db: database, private reportSaveProgress = true) {
+    constructor(private id: string, private document: document, private db: database, private reportSaveProgress: boolean, private forceRevisionCreation: boolean) {
         super();
     }
 
     execute(): JQueryPromise<saveDocumentResponseDto> {
         this.document.__metadata.id = this.id;
-
+        
         const commands: Array<Raven.Server.Documents.Handlers.BatchRequestParser.CommandData> = [
             this.document.toBulkDoc("PUT")
         ];
-        
+
         const args = ko.toJSON({ Commands: commands });
-        const url = endpoints.databases.batch.bulk_docs;
+        
+        const urlArgs = {
+           forceRevisionCreation: this.forceRevisionCreation  
+        };
+              
+        const url = endpoints.databases.batch.bulk_docs + this.urlEncodeArgs(urlArgs);
+        
         const saveTask = this.post<saveDocumentResponseDto>(url, args, this.db);
 
-        if (this.reportSaveProgress) {
+        if (this.reportSaveProgress && !this.forceRevisionCreation) {
```

Comment:
can we refactor to:
    
if (this.reportSaveProgress) {
    if (this.forceRevisionCreation) {
       ....
     } else {
        ....
     } 
}

Reply:
done